lib/logger/once.ts

```diff
@@ -0,0 +1,59 @@
+type OmitFn = (...args: any[]) => any;
+
+/**
+ * Get the single frame of this function's callers stack.
+ *
+ * @param omitFn Starting from this function, stack frames will be ignored.
+ * @returns The string containing file name, line number and column name.
+ *
+ * @example getCallSite() // => 'Object.<anonymous> (/path/to/file.js:10:15)'
+ */
+function getCallSite(omitFn: OmitFn = getCallSite): string | null {
```

Comment:
This function allows us to use it without creating any artificial keys.
We're assuming here that `filename + line + column` of the stack frame can identify particular function call.
Call stacks are expensive, so that I've tried my best to make it as effecient as possible here.

Reply:
I'm happy for it to be based on the `msg` of logger and be more efficient if necessary

Reply:
Current approach shouldn't be that fatal for performance, though. I think we can enjoy its ergonomics as far as we don't abuse it.